### PR TITLE
[Compatibility] Added GEOSEARCHSTORE command and bug fix in GEOSEARCH command

### PR DESCRIPTION
**Pull request by Vijay-Nirmal**
Adding the GEOSEARCHSTORE command to garnet and a bug fix in the response structure of the GEOSEARCH command

- [x] Add GEOSEARCHSTORE command
- [x] Add Integration Test cases, ACL Test and Slot Verification Test
- [x] Add documentation
- [x] Bug fix in the response structure of GEOSEARCH command where it returns data in unexpected RESP format
- [x] Add new test case to GEOSEARCH command using SE.Redis

### CODE REVIEW

**Comment by badrishc** [comment]
@PaulusParssinen - can you take a look at this since it may be related to your earlier work? Thanks!

**Comment by yrajas** [comment]
Thanks for adding support for more commands.
Could you please check the failing tests?

File: libs/server/Storage/Session/ObjectStore/SortedSetGeoOps.cs [hunk 1]
```diff
@@ -2,6 +2,10 @@
 // Licensed under the MIT license.
 
 using System;
+using System.Buffers;
+using System.Diagnostics;
+using System.Reflection.PortableExecutable;
```
Comment (yrajas) [inline libs/server/Storage/Session/ObjectStore/SortedSetGeoOps.cs:7]: Unused

Comment (Vijay-Nirmal) [reply libs/server/Storage/Session/ObjectStore/SortedSetGeoOps.cs]: Fixed

**Comment by Vijay-Nirmal** [comment]
@yrajas Test case failure will be fixed as part of #748

**Review by yrajas — APPROVED** [review]

**Comment by TalZaccai** [comment]
@Vijay-Nirmal Hi Vijay! Unfortunately we've noticed some issues after this PR was merged, could you please recreate it so I can review it as well?
Thanks & apologies for the inconvenience...

**Comment by Vijay-Nirmal** [comment]
@TalZaccai Created the PR #752

**Pull request merged**